musicrush/Pages/Albums/Index.cshtml.cs

```diff
@@ -25,9 +24,59 @@ public IndexModel(RazorPagesAlbumContext context)
         public SelectList Genres { get; set; }
         [BindProperty(SupportsGet = true)]
         public string AlbumGenre { get; set; }
+        public SelectList Artists {get; set;}
+        [BindProperty(SupportsGet = true)]
+        public string AlbumArtist {get; set; }
         public async Task OnGetAsync()
         {
-            Album = await _context.Album.ToListAsync();
-        }
+            //Use LINQ to get list of genres and artists.
+            IQueryable<string> genreQuery = from a in _context.Album
+                                            orderby a.Genre
+                                            select a.Genre;
+            IQueryable<string> artistQuery = from a in _context.Album
+                                            orderby a.Artist
+                                            select a.Artist;
+// public async Task OnGetAsync()
+// {
+//     // Use LINQ to get list of genres.
+//     IQueryable<string> genreQuery = from m in _context.Movie
+//                                     orderby m.Genre
+//                                     select m.Genre;
+
+//     var movies = from m in _context.Movie
+//                  select m;
+
+//     if (!string.IsNullOrEmpty(SearchString))
+//     {
+//         movies = movies.Where(s => s.Title.Contains(SearchString));
+//     }
+
+//     if (!string.IsNullOrEmpty(MovieGenre))
+//     {
+//         movies = movies.Where(x => x.Genre == MovieGenre);
+//     }
+//     Genres = new SelectList(await genreQuery.Distinct().ToListAsync());
+//     Movie = await movies.ToListAsync();
+// }            
+            var albums = from a in _context.Album
+                 select a;
+            if (!string.IsNullOrEmpty(SearchString))
+            {
+                albums = albums.Where(a => a.Title.Contains(SearchString));
+            }
+            if (!string.IsNullOrEmpty(AlbumGenre))
+            {
+                albums = albums.Where(b => b.Genre == AlbumGenre);
+            } 
+            if (!string.IsNullOrEmpty(AlbumArtist))  
+            {
+                albums = albums.Where(c => c.Artist == AlbumArtist);
+            }

```

Comment:
It looks like the user can fill in multiple selects. But in the code above, the genre will wipe out the search string, and the artist will wipe out both the search string and the genre. Is that the intent?

Reply:
My user story here was to get one to three inputs from the user and filter out the results based on those inputs. For example, if the user picks the genre of rock, all rock songs will display. Then when the user enters the artist that works in the genre, the list is further filtered, and if the artist does not work in the genre, then no results are returned. From what I tested, when all three selects are picked, the results are filtered based on all three conditions being satisfied. When only one or two selects are open, the filtering is only based on one or two conditions respectively.

Reply:
Oh, I see now. I misread what was happening. Makes sense -- thank you.